lib/global-styles.php

```diff
@@ -147,9 +157,76 @@ function gutenberg_experimental_global_styles_get_user_cpt_id() {
  * @return array Global Styles tree.
  */
 function gutenberg_experimental_global_styles_get_core() {
-	return gutenberg_experimental_global_styles_get_from_file(
+	$core_styles = gutenberg_experimental_global_styles_get_from_file(
 		dirname( dirname( __FILE__ ) ) . '/experimental-default-global-styles.json'
 	);
+	// Todo: Remove this when colors are properly set on themes.json
+	// For now colors can not be set there because json files are not translatable.
+	if ( empty( $core_styles['colors'] ) ) {
```

Comment:
We have these at the JavaScript level as well (`packages/block-editor/src/store/defaults.js`). Can we perhaps not to have them here and let the block-editor handle the defaults?

Reply:
We need to have the defaults here, in order to generate CSS vars on the frontend for each color. I think we should remove the defaults from the JS side to avoid the duplication, but as for now we only run this code if global styles are enabled it seems safer to leave the defaults there.